packages/eas-cli/src/graphql/queries/BranchQuery.ts

```diff
@@ -47,9 +50,13 @@ export const BranchQuery = {
         )
         .toPromise()
     );
-    return branch;
+    const { updateBranchByName } = response.app.byId;
+    if (!updateBranchByName) {
+      throw new BranchNotFoundError(`Could not find a branch named "${name}".`);
+    }
+    return updateBranchByName;
```

Comment:
there were roughly equal amounts of callsites that threw after `getBranchByNameAsync` did not return a branch as callsites that did not throw (3 vs 2, if i remember correctly). So I just chose to throw here, because that is generally what we do for other gql queries that don't return the expected object.

I don't feel too strongly about this, lmk if folks prefer throwing at the callsites instead.

Reply:
I think it's fine to throw here